src/Util.ts

```diff
@@ -98,9 +99,58 @@ export default class W3Util {
     });
   }
 
-  public networkGasPrice(): Promise<BigNumber> {
+  public getNetworkId(): Promise<string> {
     return new Promise((resolve, reject) => {
-      this.web3.eth.getGasPrice((e: any, r: any) => (e ? reject(e) : resolve(r)));
+      this.web3.version.getNetwork((e: any, r: any) => (e ? reject(e) : resolve(r)));
+    });
+  }
+
+  public fetchExternalApiNetworkPrice (networkId: string): Promise<any> {
+    const services = SERVICES[networkId];
+    return new Promise(async(resolve, reject) => {
+      for ( let s = 0; s < services.length; s++) {
```

Comment:
Could change these two lines to `for (let service of SERVICES[networkId]) { ... }`

Reply:
There is a preference of `for` over `for ... of`, in terms of efficiency, and we have the es-lint rule for that as well